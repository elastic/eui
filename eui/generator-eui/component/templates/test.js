import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test';

import { <%= componentName %> } from './<%= fileName %>';

describe('<%= componentName %>', () => {
  test('is rendered', () => {
    const component = render(
      <<%= componentName %> {...requiredProps} />
    );

    expect(component)
      .toMatchSnapshot();
  });
});
