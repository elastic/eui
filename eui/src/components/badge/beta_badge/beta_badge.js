import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import { EuiToolTip } from '../../tool_tip';

import {
  ICON_TYPES,
  EuiIcon,
} from '../../icon';

export const EuiBetaBadge = ({
  className,
  label,
  tooltipContent,
  tooltipPosition,
  title,
  iconType,
  ...rest,
}) => {

  const classes = classNames(
    'euiBetaBadge',
    {
      'euiBetaBadge--iconOnly': iconType,
    },
    className
  );

  let icon;
  if (iconType) {
    icon = (
      <EuiIcon
        className="euiBetaBadge__icon"
        type={iconType}
        size="m"
        aria-hidden="true"
      />
    );
  }

  if (tooltipContent) {
    return (
      <EuiToolTip
        position={tooltipPosition}
        content={tooltipContent}
        title={title || label}
      >
        <span
          className={classes}
          {...rest}
        >
          {icon || label}
        </span>
      </EuiToolTip>
    );
  } else {
    return (
      <span
        className={classes}
        title={title || label}
        {...rest}
      >
        {icon || label}
      </span>
    );
  }
};

EuiBetaBadge.propTypes = {
  className: PropTypes.string,

  /**
   * One word label like "Beta" or "Lab"
   */
  label: PropTypes.node.isRequired,

  /**
   * Supply an icon type if the badge should just be an icon
   */
  iconType: PropTypes.oneOf(ICON_TYPES),

  /**
   * Content for the tooltip
   */
  tooltipContent: PropTypes.node,

  /**
   * Custom position of the tooltip
   */
  tooltipPosition: PropTypes.string,

  /**
   * Optional title will be supplied as tooltip title or title attribute otherwise the label will be used
   */
  title: PropTypes.string,
};

EuiBetaBadge.defaultProps = {
  tooltipPosition: 'top',
};
