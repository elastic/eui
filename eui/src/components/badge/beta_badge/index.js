export {
  EuiBetaBadge,
} from './beta_badge';
