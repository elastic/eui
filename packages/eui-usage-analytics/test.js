/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

const Codeowners = require('codeowners');
const temp = new Codeowners("../../../kibana");

const { scan } = require('./scan');

const runScan = async () => {
  const scanResult = await scan();
  console.log(scanResult);
};

runScan();
