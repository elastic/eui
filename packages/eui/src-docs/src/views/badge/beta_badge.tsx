import React from 'react';
import { css } from '@emotion/react';

import { EuiBetaBadge, EuiSpacer, EuiTitle } from '../../../../src/components';

const defaultColors = ['hollow', 'accent'] as const;
const deprecationColors = ['subdued', 'warning'] as const;

const wrapperStyles = css`
  display: grid;
  align-items: flex-start;
  gap: 1rem;
`;

export default () => (
  <>
    <EuiTitle size="xs">
      <h3>Colors and types</h3>
    </EuiTitle>
    <EuiSpacer size="m" />
    <div
      css={[
        wrapperStyles,
        css`
          grid: repeat(2, max-content) / repeat(6, max-content);
        `,
      ]}
    >
      {defaultColors.map((item) => {
        const label = 'Beta';
        return (
          <>
            <EuiBetaBadge
              label={label}
              color={item}
              tooltipContent="This module is not GA. Please help us by reporting any bugs."
            />
            <EuiBetaBadge
              label={label}
              color={item}
              size="s"
              tooltipContent="This module is not GA. Please help us by reporting any bugs."
            />
            <EuiBetaBadge label={label} color={item} iconType="beta" />
            <EuiBetaBadge label="Tech preview" color={item} iconType="beaker" />
            <EuiBetaBadge size="s" label={label} color={item} iconType="beta" />
            <EuiBetaBadge
              size="s"
              label="Tech preview"
              color={item}
              iconType="beaker"
            />
          </>
        );
      })}
    </div>
    <EuiSpacer />
    <div
      css={[
        wrapperStyles,
        css`
          grid: repeat(2, max-content) / repeat(2, max-content);
        `,
      ]}
    >
      {deprecationColors.map((item) => {
        const label = 'Deprecated';
        return (
          <>
            <EuiBetaBadge
              label={label}
              color={item}
              tooltipContent="This module is not GA. Please help us by reporting any bugs."
            />
            <EuiBetaBadge
              label={label}
              color={item}
              size="s"
              tooltipContent="This module is not GA. Please help us by reporting any bugs."
            />
          </>
        );
      })}
    </div>
    <EuiSpacer />
    <EuiTitle size="xs">
      <h3>Clickable example</h3>
    </EuiTitle>
    <EuiSpacer size="m" />
    <div>
      <EuiBetaBadge
        label="Beta"
        href="http://www.elastic.co/subscriptions"
        target="_blank"
      />
    </div>
    <EuiSpacer />
    <EuiTitle size="xs">
      <h3>Placement alongside titles</h3>
    </EuiTitle>
    <EuiSpacer size="m" />
    <EuiTitle size="s">
      <h3>
        New feature &nbsp;
        <EuiBetaBadge
          label="Beta"
          size="s"
          tooltipContent="This module is not GA. Please help us by reporting any bugs."
        />
      </h3>
    </EuiTitle>
  </>
);
