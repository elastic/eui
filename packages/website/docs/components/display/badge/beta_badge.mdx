---
slug: /display/badge/beta
id: display_badge_beta
---

# Beta badge

The **EuiBetaBadge** was created specifically to call out modules that are not in GA. Generally the labels used are "Beta" or "Lab". They require an extra `tooltipContent` to describe the purpose of the badge. You can pass an optional `title` prop to populate the tooltip title or html title attribute but by default it will use the `label`.

If you pass in an `iconType`, only the icon will be used in the badge itself and the label will be applied as the title. Only use an icon when attaching the beta badge to small components. Beta badges can also be made clickable by passing `href` or `onClick` as needed.

They can also be used in conjunction with [**EuiCards**](/docs/display/card) and [**EuiKeyPadMenuItems**](/docs/navigation/key-pad-menu).

```tsx interactive
import React from 'react';
import { EuiBetaBadge, EuiSpacer, EuiTitle } from '@elastic/eui';
import { css } from '@emotion/react';

const defaultColors = ['hollow', 'accent'] as const;
const deprecationColors = ['subdued', 'warning'] as const;

const wrapperStyles = css`
  display: grid;
  align-items: flex-start;
  gap: 1rem;
`;

export default () => (
  <>
    <EuiTitle size="xs">
      <h3>Colors and types</h3>
    </EuiTitle>
    <EuiSpacer size="m" />
    <div
      css={[
        wrapperStyles,
        css`
          grid: repeat(2, max-content) / repeat(6, max-content);
        `,
      ]}
    >
      {defaultColors.map((item) => {
        const label = 'Beta';
        return (
          <>
            <EuiBetaBadge
              label={label}
              color={item}
              tooltipContent="This module is not GA. Please help us by reporting any bugs."
            />
            <EuiBetaBadge
              label={label}
              color={item}
              size="s"
              tooltipContent="This module is not GA. Please help us by reporting any bugs."
            />
            <EuiBetaBadge label={label} color={item} iconType="beta" />
            <EuiBetaBadge label="Tech preview" color={item} iconType="beaker" />
            <EuiBetaBadge size="s" label={label} color={item} iconType="beta" />
            <EuiBetaBadge
              size="s"
              label="Tech preview"
              color={item}
              iconType="beaker"
            />
          </>
        );
      })}
    </div>
    <EuiSpacer />
    <div
      css={[
        wrapperStyles,
        css`
          grid: repeat(2, max-content) / repeat(2, max-content);
        `,
      ]}
    >
      {deprecationColors.map((item) => {
        const label = 'Deprecated';
        return (
          <>
            <EuiBetaBadge
              label={label}
              color={item}
              tooltipContent="This module is not GA. Please help us by reporting any bugs."
            />
            <EuiBetaBadge
              label={label}
              color={item}
              size="s"
              tooltipContent="This module is not GA. Please help us by reporting any bugs."
            />
          </>
        );
      })}
    </div>
    <EuiSpacer />
    <EuiTitle size="xs">
      <h3>Clickable example</h3>
    </EuiTitle>
    <EuiSpacer size="m" />
    <div>
      <EuiBetaBadge
        label="Beta"
        href="http://www.elastic.co/subscriptions"
        target="_blank"
      />
    </div>
    <EuiSpacer />
    <EuiTitle size="xs">
      <h3>Placement alongside titles</h3>
    </EuiTitle>
    <EuiSpacer size="m" />
    <EuiTitle size="s">
      <h3>
        New feature &nbsp;
        <EuiBetaBadge
          label="Beta"
          size="s"
          tooltipContent="This module is not GA. Please help us by reporting any bugs."
        />
      </h3>
    </EuiTitle>
  </>
);

```

## Props

import docgen from '@elastic/eui-docgen/dist/components/badge';

<PropTable definition={docgen.EuiBetaBadge} />
