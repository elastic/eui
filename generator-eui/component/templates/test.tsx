/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React from 'react';
import { render } from '../../test/rtl';
import { requiredProps } from '../../test/required_props';

import { <%= componentName %> } from './<%= fileName %>';

describe('<%= componentName %>', () => {
  it('renders', () => {
    const { container } = render(
      <<%= componentName %> {...requiredProps} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
