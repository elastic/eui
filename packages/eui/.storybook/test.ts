/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import { queries, within, waitFor, fireEvent, expect } from '@storybook/test';
import * as dataTestSubjQueries from '../src/test/rtl/data_test_subj_queries';

/**
 * Custom Storybook within util with EUI query helpers
 * + additional chained async/waitFor component utils
 *
 * @see https://storybook.js.org/docs/writing-stories/play-function#writing-stories-with-the-play-function
 * @see https://testing-library.com/docs/dom-testing-library/api-within/
 */
const customWithin = (canvasElement: HTMLElement) => {
  const canvas = within<typeof queries & typeof dataTestSubjQueries>(
    canvasElement,
    { ...queries, ...dataTestSubjQueries }
  );

  return {
    ...canvas,

    /**
     * 1. Loki doesn't like userEvent, only fireEvent
     * 2. Storybook fires fireEvents too early (esp. on page load), so we add a waitFor
     */
    waitForAndClick: async (testSubject: string) => {
      await waitFor(() =>
        expect(canvas.getByTestSubject(testSubject)).toBeInTheDocument()
      );
      await fireEvent.click(canvas.getByTestSubject(testSubject));
    },

    waitForEuiPopoverVisible: async () =>
      await waitFor(() =>
        expect(canvasElement.querySelector('[data-popover-open]')).toBeVisible()
      ),
    waitForEuiPopoverHidden: async () =>
      await waitFor(() =>
        expect(
          canvasElement.querySelector('[data-popover-panel]')
        ).not.toBeInTheDocument()
      ),
  };
};

export { customWithin as within };
