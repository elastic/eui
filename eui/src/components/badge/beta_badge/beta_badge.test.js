import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../../test';

import { EuiBetaBadge } from './beta_badge';

describe('EuiBetaBadge', () => {
  test('is rendered', () => {
    const component = render(
      <EuiBetaBadge label="Beta" {...requiredProps} />
    );

    expect(component)
      .toMatchSnapshot();
  });
});
