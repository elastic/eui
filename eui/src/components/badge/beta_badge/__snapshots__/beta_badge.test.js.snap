// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiBetaBadge is rendered 1`] = `
<span
  aria-label="aria-label"
  class="euiBetaBadge testClass1 testClass2"
  data-test-subj="test subject string"
  title="Beta"
>
  Beta
</span>
`;
